addons/arrays/fnc_prettyPrint.sqf

```diff
@@ -0,0 +1,67 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_prettyPrint
+
+Description:
+	Makes an array easier to read.
+
+Parameters:
+    _this - Array <ARRAY>
+
+Returns:
+	Formatted string <STRING>
+
+Examples:
+    (begin example)
+		_return = [0, 1, ["22", 33, []], 4] call CBA_fnc_prettyPrint;
+		// _return ==>
+		// "[
+		//		0,
+		//		1,
+		//		[
+		//			""22"",
+		//			33,
+		//			[]
+		//		],
+		//		4
+		//	]" 
+    (end)
+
+Author:
+   Terra 
+---------------------------------------------------------------------------- */
+SCRIPT(prettyPrint);
+
+if (count _this == 0) exitWith {
+	"[]"
+};
+private _lines = ["["];
+// Recursive function calls have _depth already defined
```

Comment:
This means that the function behaves differently if _depth already exists. This is very bad design wise. I recommend making _depth an explict function argument instead. Could use a local function if calling signature of this function should be kept, but also consider changing that, since passing arrays is dodgy too.